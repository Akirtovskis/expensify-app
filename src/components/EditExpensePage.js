import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense} from '../actions/expenses';

//Refactor EditExpense to class based component
//Setup mapDispatchToProps editExpense removeExpense

//beforeEach()
//Should render edit expense page (SNAPSHOT)
//Should handle edit expense (SPIES)
//Should handle remove expense (REMOVE)
export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onRemove = (expense) => {
    this.props.startRemoveExpense({id:this.props.expense.id});
    this.props.history.push('/');
  }
  render(){
    return (
      <div>
        <ExpenseForm
          expense = {this.props.expense}
          onSubmit = {this.onSubmit}
        />
        <button onClick={this.onRemove}
        >Remove</button>
      </div>
    );
  };
}

const mapStateToProps = (state, props) => {
  return {
    expense : state.expenses.find((expense)=> expense.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense : (id, expense)=>dispatch(editExpense(id, expense)),
  startRemoveExpense : (data)=>dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
