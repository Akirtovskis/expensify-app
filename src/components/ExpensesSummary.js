import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';


export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount===1 ? ' expense ' : ' expenses ';
  const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing : <span>{expenseCount}</span> {expenseWord} totalling : <span>{formattedExpensesTotal}</span>
        </h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

// Here connecting component to store! NB (SHOULD look for ways to descructure props)
const mapStateToProps = (state)=> {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount : visibleExpenses.length,
    expensesTotal : selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
