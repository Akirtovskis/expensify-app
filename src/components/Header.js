import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <h3>
      <NavLink to='/' activeClassName='is-active' exact={true}>Home </NavLink>
      <NavLink to='/create' activeClassName ='is-active'>Create Expense </NavLink>
    
    </h3>
  </header>
);

export default Header;
