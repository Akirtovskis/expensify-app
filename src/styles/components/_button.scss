.button {
  background:$blue;
  border-radius:3px;
  color:white;
  display:inline-block;
  font-size:$font-size-large;
  font-weight:300;
  line-height:1;
  padding:$s-size;
  text-decoration:none;
}

.button--logout {
  background:none;
  border:none;
}

.button--secondary {
  background:#888888;
}
