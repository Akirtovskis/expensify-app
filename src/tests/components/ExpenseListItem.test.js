import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpenseListItem } from '../../components/ExpenseListItem';


test('Should render the individual expense with provided data', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
  expect(wrapper).toMatchSnapshot();
})
