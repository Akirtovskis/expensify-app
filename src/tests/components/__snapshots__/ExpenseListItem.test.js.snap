// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the individual expense with provided data 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Gum
    </h3>
    <span
      className="list-item__sub-title"
    >
      January 1st, 1970
    </span>
  </div>
  <h3
    className="list_item__data"
  >
    $1.95
  </h3>
</Link>
`;
